virtualcluster/pkg/syncer/resources/pod/dws.go

```diff
@@ -64,6 +64,12 @@ func (c *controller) Reconcile(request reconciler.Request) (res reconciler.Resul
 	if err := c.MultiClusterController.Get(request.ClusterName, request.Namespace, request.Name, vPod); err != nil && !apierrors.IsNotFound(err) {
 		return reconciler.Result{Requeue: true}, err
 	}
+	// if LabelTenantSchedulerName is specified, bypass syncing
+	schedulerlabel, ok := vPod.GetLabels()[constants.LabelTenantSchedulerName]
+	if ok && schedulerlabel != "" {
+		klog.V(5).Infof("skip syncing pod with tenant scheduler label")
+		return reconciler.Result{}, nil
+	}
```

Comment:
If you add this logic - https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/main/virtualcluster/pkg/syncer/resources/pod/controller.go#L317-L319 it only even enqueue those pods so this check won't be necessary to fire, instead the informer will ignore it, it also needs to be behind a feature gate IMO.